python/tester/upload.py

```diff
@@ -84,34 +153,222 @@ def main():
         "./prompt-proto-upload.json"
     )
     storage_client = storage.Client(PROJECT_ID, credentials=credentials)
-    bucket = storage_client.bucket("rubin-prompt-proto-main")
+    dest_bucket = storage_client.bucket("rubin-prompt-proto-main")
     batch_settings = pubsub_v1.types.BatchSettings(
         max_messages=INSTRUMENTS[instrument].n_detectors,
     )
     publisher = pubsub_v1.PublisherClient(credentials=credentials,
                                           batch_settings=batch_settings)
 
+    last_group = get_last_group(storage_client, instrument, date)
+    _log.info(f"Last group {last_group}")
+
+    src_bucket = storage_client.bucket("rubin-prompt-proto-unobserved")
+    raw_pool = get_samples(src_bucket, instrument)
+
+    if raw_pool:
```

Comment:
I kind of want the two parts of this `if` to be separate functions: `handle_real_raws` and `handle_random_raws` or something like that?